radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file RPdb.c
+ * @date February 21, 2022
+ * @author Isaac Poirier (iap992)
+ */
+
+#include <RPdb.h>
+#include <RI2c.h>
+#include <string.h>
+
+
+
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
+/**
+ * Conversion Constants for between ADC Values and Real Values
+ */
+
+/** Used in: (Volts = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_VOLTAGE			((float) 0.008993157)
+#define ADC_COUNT_TO_BCR_VOLTAGE				((float) 0.0322581)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE	((float) 0.008978)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE		((float) 0.005865)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE		((float) 0.004311)
+
+
+/** Used in: (mAmps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_CURRENT			((float) 14.662757)
+
+/** Used in: (Amps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_AMPERAGE				((float) 0.0009775)
+#define ADC_COUNT_TO_3V3_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_5V_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_CURRENT	((float) 0.005237)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_CURRENT		((float) 0.005237)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_CURRENT		((float) 0.005237)
+
+
+/** Used in: (Temp in Deg C = (Scale * ADC Count) - Shift) */
+#define ADC_COUNT_TO_SOLAR_ARRAY_TEMP_SCALE	((float) 0.4963)
+// Can use this \/ interchangeably between solar array and motherboard temp shift
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT ((float) 273.15)
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE ((float) 0.372434)
+
+
+/** Used in: (Irradiance = Constant * ADC Count) */
+#define ADC_COUNT_TO_IRRADIANCE				((float) 1.59725)
```

Comment:
What is irradiance, I have never heard that term before

Reply:
W/(m^2) is the units they give me as a return for the sun detector telemetry. I didn't know what this unit was either so I googled and Irradiance is what came up for it